Add words about the need for a web browser

andreww opened the pull request:
Neither IPython notebooks nor github.com will
work well with old web browsers. This commit
adds text to the start of the setup section
asking participants to make sure they have an
up to date web browser (linking to the list
of supported browsers for github and the
IPython notebook). We also sugest that the
OS should be updated.

This fixes #104, but the exact wording may
need a tweak.

andreww committed:
Add words about the need for a web browser

Neither IPython notebooks nor github.com will
work well with old web browsers. This commit
adds text to the start of the setup section
asking participants to make sure they have an
up to date web browser (linking to the list
of supported browsers for github and the
IPython notebook). We also sugest that the
OS should be updated.

This fixes #104, but the exact wording may
need a tweak.

rgaiacs commented on index.html:
```diff
@@ -265,6 +265,14 @@
 
 <p>
+  To participate in a Software Carpentry workshop, you will need 
+  access to the software described below. In addition, you will 
```
Instead of a note here I prefer to add the web browser to the description of the others softwares.

rgaiacs commented on index.html:
```diff
@@ -265,6 +265,14 @@
+  To participate in a Software Carpentry workshop, you will need 
+  access to the software described below. In addition, you will 
+  need a web browser 
+  <a href='https://help.github.com/articles/supported-browsers/'
```
Instead of point the students to GitHub and Jupyter compatibility page I prefer "the last stable version of your favorite web browser".

rgaiacs commented on index.html:
```diff
@@ -265,6 +265,14 @@
+  <a href='https://help.github.com/articles/supported-browsers/'
+  >able to access github.com</a> and run the 
+  <a href='http://ipython.org/ipython-doc/2/install/install.html#browser-compatibility'
+  >IPython notebook</a>. It is probably worth updating your operating 
```
"It is probably worth updating your operating system" sounds like that I need to replace Windows 7 with Windows 8, replace OS X Mountain Lion wit OS X Yosemite or Ubuntu 14.04 with Ubuntu 14.10. I'm -1 to this.

gvwilson commented:
1.  I agree we should tell people they need a modern enough web browser.
2.  We should add a test for this to `setup/swc-installation-test-2.py` (@wking may be able to help).
3.  We shouldn't suggest that they have to have the IPython Notebook installed - R and MATLAB workshops won't.

wking commented:
On Sun, Dec 28, 2014 at 11:32:49AM -0800, Greg Wilson wrote:

> 1.  We should add a test for this to
>    `setup/swc-installation-test-2.py`

It already checks for a browser [1,2,3,4].  I don't have minimum
versions for the different browsers at the moment, but if you tell me
what they should be I can add them.  I'm also not sure how to extract
the Safari version number from Python, so if you want a minimum Safari
version we'll need someone to figure out how that works.

andreww committed:
Update web browser instructions.

This moves the requierments into the relevent
"section" (git and python), and lists the supported
versions. This way e.g. the Python requierment will
go away for an R workshop.

I've left a generic "you will need a web browser" at
the top of the insstallation instructions. People will
need some sort of browser to access an Etherpad and any
documentation.

Links to the requierments page are in comments (so that
these can easily be checked and updated in fututre).

andreww commented:
Apologies for the delay on this. January arrived with two grant deadlines and a workshop.

Does this version (0a636bbe72fd75bd74c7e096f703f3cb0b49597b) look better? I've put explicit supported version numbers in the Python and Git blocks so they will go away for an R and mercurial workshop (@gvwilson's point 3). I've also removed reference to an OS update and moved the links to the requirements to comments (which should make update easer).

@wking - for the browsers you check for, the version numbers should be > 6 for Firefox, >13 for Chrome and >5 for Safari. But these ought only to apply to Python workshops, Github supports "the current" versions of these three plus IE >9.

rgaiacs commented on index.html:
```diff
@@ -428,6 +437,13 @@ <h2 id="setup">Setup</h2>
+  <p>
+    We will teach Python using the IPython notebook, a programming environment
+    that runs in a web browser. For this to work you will need a reasonably
+    up-to-date browser. The Chrome (version 13 and above), Safari (version 5
```
This versions of Chrome and Firefox is very old!

gvwilson commented:
@andreww can you please update this in the next couple of days so that we can merge it?

andreww committed:
Remove version numbers from Python web browsers

Rather than list the exact version numbers of
"supported" browsers needed to run IPython notebooks,
just say that the current versions are supported
(which is true) and that some older versions are not
(which is also true). The actual constraint (for everything
other than IE) will come from GitHub anyway (which only
supports "current versions" of these browsers, but
supports IE 9).

andreww commented:
Is 69272cd8ac3cfb64e84662ccb202a0138cc3aaaa OK for the text? We should probably open a new issue for the test scripts.

wking commented:
On Tue, Jan 20, 2015 at 02:30:31AM -0800, Andrew Walker wrote:

> for the browsers you check for, the version numbers should be > 6
> for Firefox, >13 for Chrome and >5 for Safari.

I've added checks for Firefox, Chrome, and Chromium with d13b26c9
(swc-installation-test-2.py: Add minimum browser requirements for
IPython, 2015-01-24, [1,2]).  If these look good, I'll cut an
installation-test release and update my bc-namespaced branch, and you
can pull the changes into this repository.

As I pointed out earlier, I'm not aware of a command-line method for
figuring out the Safari version number, so the script is currently
assuming that any Safari install will be new enough.

> But these ought only to apply to Python workshops, Github supports
> "the current" versions of these three plus IE >9.

It looks like GitHub also tries to support the ESR version of Firefox
[3](https://help.github.com/articles/supported-browsers/#firefox-extended-support-release), and that's currently 31.4.0 [4](https://www.mozilla.org/en-US/firefox/31.4.0/releasenotes/).  I'm running 24.8.0, and haven't
noticed any recent problems using GitHub, so I'm not sure if we want
to try and keep track of what GitHub actually supports, or just bump
the minimum version whenever someone reports a bug with a given
browser version on GitHub.  I don't know how we'd test to see if
GitHub fixed a given issue though (restoring support for the older
browser).  In October I reported that the fork button was broken with
Firefox 24.  GitHub was only officially supporting Firefox 32 (the
then-latest stable), but a few days later fork started working for me
again on Firefox 24.  If GitHub posted reliable compatibility notes
for older browsers, I'd be fine checking for them, but with their
current phrasing I don't think it's worth our time to bump whenever a
browser releases a new stable version (matching GitHub's offical
compatibility) or guessing about what older versions are actually
working (which would allow folks with older, working browsers like
mine to avoid upgrading before their distro stabled a more recent
version).

wking commented:
On Sat, Jan 24, 2015 at 11:42:37AM -0800, W. Trevor King wrote:

> As I pointed out earlier, I'm not aware of a command-line method for
> figuring out the Safari version number, so the script is currently
> assuming that any Safari install will be new enough.

Greg asked after this on Twitter [1](https://twitter.com/gvwilson/status/559089567559843841), and Philip Guo suggested using
version.plist [2](https://twitter.com/pgbovine/status/559094439009075203).  I've sketched this out, but without an OS X box to
test on I'm not sure it works ;).  Can someone run the tweaked script
[3,4] through its paces with something like:

  $ ./swc-installation-test-2.py sublime-text textmate textwrangler safari xcode

I don't know if anyone will have all those editors installed, but it
would be nice to know how many of those version checks work.

andreww commented:
> I don't know if anyone will have all those editors installed, but it would be nice to know how many of those version checks work.

@wking Once I fixed the minor syntax issue that stopped the code running (see wking/swc-setup-installation-test#1) this ran fine for me and found my safari version. This is OSX 10.6.8 so if somebody with a more modern system (or some text editors) could test this too that would be useful.

gvwilson commented on index.html:
```diff
@@ -428,6 +437,14 @@ <h2 id="setup">Setup</h2>
+    Firefox browsers are all supported (some older browsers, including Internet Explorer
+    version 9 and below, are not).
+  </p>
+
```
We don't always teach with the Notebook (in fact, I never do any more with novices - too many new things at once).

andreww replied:
As it says in the HTML comment a few lines above: "Remove the second paragraph if the workshop will teach Python using something other than the IPython notebook." 

Should this be moved closer to the paragraph in question? I guess it is more common to use IPython than not (and thus fewer people will need to remove this than add it if we did not include it from the template).

gvwilson commented:
@tomwright01 could you please have a look at this one?

tomwright01 commented on index.html:
```diff
@@ -418,7 +424,10 @@ <h2 id="setup">Setup</h2>
 </div> <!-- End of 'Git' section. -->
 
-<div> <!-- Start of 'Python' section. -->
+<div> <!-- Start of 'Python' section. Remove the second paragrapph if
```
sp. paragraph

tomwright01 commented on index.html:
```diff
@@ -375,14 +378,17 @@ <h2 id="setup">Setup</h2>
     version of your code
-    on <a href="https://github.com/">github.com</a>.
+    on <a href="https://github.com/">github.com</a>. You will need a
+    supported web browser (current versions of Chrome, Firefox or Safari,
```
Can we convert 'supported' into a hyperlink to the supported-browsers webpage

tomwright01 commented on index.html:
```diff
@@ -428,6 +437,14 @@ <h2 id="setup">Setup</h2>
+    We will teach Python using the IPython notebook, a programming environment
+    that runs in a web browser. For this to work you will need a reasonably
+    up-to-date browser. The current versions of the Chrome, Safari and
+    Firefox browsers are all supported (some older browsers, including Internet Explorer
```
add hyperlink to browser-compatibility webpage

wking commented:
On Mon, Jan 26, 2015 at 11:04:50AM -0800, Andrew Walker wrote:

> > I don't know if anyone will have all those editors installed, but
> > it would be nice to know how many of those version checks work.
> 
> @wking Once I fixed the minor syntax issue that stopped the code
> running (see wking/swc-setup-installation-test#1) this ran fine for
> me and found my safari version. This is OSX 10.6.8 so if somebody
> with a more modern system (or some text editors) could test this too
> that would be useful.

I gave up on further testing and just merged the branch into
swc-setup-installation-test's master [1](https://github.com/wking/swc-setup-installation-test/commit/33574ab399e537250b16d6924fe00a3a9ebf1a3b) and bc-namespaced [2](https://github.com/wking/swc-setup-installation-test/commit/b36372f583ff7bb9b254101794069b2f31b5401a).  If
the workshop-maintainers are comfortable with the changes, I'd
suggest they run:

  $ git pull --log git://github.com/wking/swc-setup-installation-test bc-namespaced

which only conflicts due to the README.md → index.md and subsequent
local edits that have been made locally (see my comments in #86 and
#89 for details).  The commits comming in are currently:

  Fix decorator in VersionPlistCommandDependency
  swc-installation-test-2.py: Convert PathCommandDependency to VersionPlistCommandDependency
  swc-installation-test-2.py: Add minimum browser requirements for IPython
  swc-installation-test-2.py: Collect classes together
  README.md: Change headers ("Students" -> "For leaners")

Since the README changes are already part of index.md, I'd resolve and
complete the merge with:

  $ git rm setup/README.md
    $ git commit

And then push to publish the change to this repository.

andreww committed:
Add hyperlinks to supported pages

As sugested by @tomwright01, add hyperlinks to
the list of supported web pages for github and
IPython.

Also fix the paragrapph -> paragraph typo.

andreww commented:
@tomwright01 - I've added the hyperlinks and fixed the typo. I've also rebased the branch on top of the more recent changes to gh-pages. OK to merge?

wking commented:
On Mon, Mar 16, 2015 at 06:59:07AM -0700, Tom Wright wrote:

> Merged #108.

It doesn't look like anyone's pulled in my associated
installation-test changes (suggested command in [1](https://github.com/swcarpentry/workshop-template/pull/108#issuecomment-77439680)).  I'm happy to do
it myself if a workshop-template maintainer green-lights the change.
